src/dbt_score/models.py

```diff
@@ -561,6 +566,17 @@ def _reindex_tests(self) -> None:
                 ):
                     self.tests[node_unique_id].append(node_values)
 
+    def _populate_parents(self) -> None:
+        """Populate `parents` for all models and snapshots."""
+        for node in list(self.models.values()) + list(self.snapshots.values()):
+            for parent_id in node.depends_on.get("nodes", []):
+                if parent_id in self.models:
+                    node.parents.append(self.models[parent_id])
+                elif parent_id in self.snapshots:
+                    node.parents.append(self.snapshots[parent_id])
+                elif parent_id in self.sources:
+                    node.parents.append(self.sources[parent_id])
```

Comment:
```suggestion
 def _populate_parents(self) -> None:
    """Populate `parents` for all models and snapshots."""
    all_parents = {**self.models, **self.snapshots, **self.sources}
    for node in list(self.models.values()) + list(self.snapshots.values()):
        for parent_id in node.depends_on.get("nodes", []):
            if parent := all_parents.get(parent_id):
                node.parents.append(parent)
```

IMO, this looks easier to read and maintain but just a suggestion :)